bq-connector/docai_bq_connector/bigquery/StorageManager.py

```diff
@@ -63,6 +63,47 @@ def write_record(self, table_id: str, record):
             logging.error("Encountered errors while inserting rows: %s", errors)
         return errors
 
+    def update_record(self, table_id: str, record_id_name, record_id_value, cols_to_update):
+        # Assumes all columns and key are of type STRING
+        query_params = []
+        dml_statement = f"UPDATE `{self.project_id}.{self.dataset_id}.{table_id}` SET"
+        if len(cols_to_update) == 0:
+            # Nothing to do
+            return
+        idx = 0
+        for cur_col, cur_val in cols_to_update.items():
+            dml_statement = f"{dml_statement} {cur_col} = @param_{idx},"
+            cur_qp = bigquery.ScalarQueryParameter(f'param_{idx}', 'STRING', cur_val)
+            query_params.append(cur_qp)
+            idx += 1
+        # Remove last comma
+        dml_statement = f"{dml_statement[:-1]} WHERE {record_id_name} = @param_record_id"
+        cur_qp = bigquery.ScalarQueryParameter('param_record_id', 'STRING', record_id_value)
+        query_params.append(cur_qp)
+        logging.debug(f"About to run query: {dml_statement} with params: {query_params}")
+        query_job_config = bigquery.QueryJobConfig(use_legacy_sql=False)
```

Comment:
You should be able to put the query parameters as an argument into the constructor.
It's a bit cleaner than adding it in another command.

Reply:
Done